haskell/haddock.bzl

```diff
@@ -326,7 +333,10 @@ haskell_doc = rule(
     _haskell_doc_rule_impl,
     attrs = {
         "deps": attr.label_list(
-            aspects = [haskell_doc_aspect],
+            aspects = [
+                haskell_cc_libraries_aspect,
```

Comment:
Do we really need to access the cc libraries here?

Reply:
Yes, the haddock aspect depends on the C libraries [here](https://github.com/tweag/rules_haskell/blob/513e775f00fcddbc5d2fc1a62e6818e7ea5025f0/haskell/haddock.bzl#L136). If we don't add the `haskell_cc_libraries_aspect` here, then that aspect will fail to find the `HaskellCcLibrariesInfo` provider. E.g. for `//tests:toto`
```
<target //tests:utils> (rule '_haskell_library') doesn't contain declared provider 'HaskellCcLibrariesInfo'
```

Reply:
Oh indeed, should have looked better. Now I wonder why haddock needs these, but that's a whole other topic

Reply:
No worries, it's a good question.